app/components/pages/SubmissionWizard/steps/Editors/EditorsContent.js

```diff
@@ -37,6 +36,23 @@ const OptionalExclude = ({
       ?
     </Box>
   )
+/*
+ * If using the OptionalExclude component you need to extend the function
+ * below, to define when its visible.
+ */
+const OptionalExcludeVisibilites = (nextProps, prevState) => ({
```

Comment:
I'm not sure I understand why this has been abstracted out from `getDerivedStateFromProps`?

Reply:
This has been done to make the code more maintainable

Reply:
Yeah I see that the comment above says this makes it extendable. I'm not necessarily saying I disagree. I just genuinely don't understand.

What scenario does this cater for, that couldn't be done by adding directly into `getDerivedStateFromProps`?

Reply:
Let's talk about it IRL ... it basically explicitly ties the functionality of `getDerivedStateFromProps` to the `OptionalExclude` component by name. I'd like this done my making a class and using a design pattern for state management - as it terns out React makes this complicated. Would like to talk to Tam about it - incase he knows any React patterns that would make this pretty.

Reply:
Yeah we can definitely talk about it IRL if you prefer. I was just wondering about the reasoning, so that I could tell whether to pop this into the implementation I'm working on?

From what I can tell, your intention for this is about making state manipulation class-based? But this feels a little like fighting against React itself, since the premise of the library is that UI is conceived of as a function of state. So by design, the manipulation of state is intended to remain functional.

Happy to talk about changes like this with @tamlyn when he comes back, but for now I would suggest that we stay with a more functional approach?